src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -375,8 +373,7 @@ private SortedSet<Violation> processFile(File file) throws CheckstyleException {
      * @return {@code true} if the file is accepted.
      */
     private boolean acceptFileStarted(String fileName) {
-        final String stripped = CommonUtil.relativizePath(basedir, fileName);
-        return beforeExecutionFileFilters.accept(stripped);
```

Comment:
this is anyway an violation fixed in https://github.com/checkstyle/checkstyle/pull/16488 Issue #16421: Local variables should not be declared and then immediately returned or thrown - (RSPEC-S1488)